### PR TITLE
Peerprep Submissions History (N2)

### PR DESCRIPTION
**Pull request by cadencjk**
This PR implements the following:

1) Set up History Microservice
    - Listen on Port 9001
    - Add submission attempt to DB with API call
    - Delete all submissions belonging to a user with API call
    - Retrieve all submissions belonging to a user with API call
    - Add another attribute in DB to store the code the user typed
    
2) Frontend
    - Add "Submit" button in Collaboration page
        - (userId, questionId, code) will be sent to backend which still store (id, userId, questionId, timeStamp, code) in attempts table
    - Add submission history table
    - Map questionId to question title in submission history table

TODO:
1) Need to update History DB whenever there is a deletion of a particular user / question
2) Implement JWT check for History microservice

### CODE REVIEW

File: History/HistoryRepository.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
+const mysql = require('mysql2');
+const env = require('./loadEnvironment.js');
+
+const conn = mysql.createConnection({
+  ...env.mysqlCreds,
+  ...{ database: env.mysqlDbName },
+});
+
+const findAttemptByEmailAndQuestion = async (userId, questionId) => {
```
Comment (venuslimm) [inline History/HistoryRepository.js:9]:
```suggestion
const findAttemptByUserIdAndQuestionId = async (userId, questionId) => {
```

File: History/HistoryController.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+const deleteUserAttempts = async (req, res) => {
+  try {
+    const { userId } = req.body;
+    await historyService.deleteUserAttempts(userId);
+    res.json({ message: 'SUCCESS: History attempts deleted' });
+  } catch (err) {
+    res.status(err?.status || 400).json({ error: err?.message || err });
+  }
+};
```
Comment (venuslimm) [inline History/HistoryController.js:21]: Should user be allowed to delete their attempts?

File: History/HistoryRepository.js (hunk 1 above)
Comment (cadencjk) [reply History/HistoryRepository.js]: Good catch! Resolved

File: History/HistoryController.js (hunk 2 above)
Comment (cadencjk) [reply History/HistoryController.js:21]: This is to prepare when a user deregisters their account. Then all records belonging to the user should also be wiped off.

User service will send a notification for History service to call this function.

**Pull request merged**